rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -570,16 +570,32 @@ public Host(InetAddress address, int nodesNumber) {
                 this.nodesNumber = nodesNumber;
             }
 
+            @Override
+            public int hashCode() {
+                final int prime = 31;
+                int result = 1;
+                result = prime * result + getOuterType().hashCode();
+                result = prime * result + ((address == null) ? 0 : address.hashCode());
+                return result;
+            }
+
+            @Override
             public boolean equals(Object obj) {
-                if (obj instanceof Host) {
-                    Host host = (Host) obj;
-                    if (address == null && host.address == null) {
-                        return true;
-                    } else if (address != null && host.address != null) {
-                        return address.equals(host.address);
-                    }
-                }
-                return false;
+                if (this == obj)
+                    return true;
+                if (obj == null)
+                    return false;
+                if (getClass() != obj.getClass())
+                    return false;
+                Host other = (Host) obj;
+                if (!getOuterType().equals(other.getOuterType()))
```

Comment:
not sure that the outerType is needed as we are trying to compare two Host instances